### PR TITLE
Items Embedding in Limbs

### PR DESCRIPTION
**Pull request by RemieRichards**
- Explosions throw items
- Items can embed in limbs when <B>thrown</B>
- Embedded items cause damage
- Embedded items cause increased bleeding based on (0.5*amount_of_objects_in_limb)
- Embedded items can fall out, causing damage but removing the item
- Embedded items may be removed <B>ONE AT A TIME</B> by the person they are embedded in at the cost of some damage
- Surgery for embedded item removal (Removes ALL objects embedded in that same limb)
- Throwing Stars! (Uplink item, 100% chance to embed, 0% chance to fall out)
- Cleans up is_sharp() a little bit
- Cleans up help_shake_act() a little bit

<B>Embedding:</B>
embedding
<B>Embedded item removal:</B>
removingrods

Also I added a changelog in the first commit! I'm adapting to our new ways!

### CODE REVIEW

**Comment by Thunder12345** [comment]
Now we just need to be able to jam ball bearings/screwdrivers/glass shards in grenades

**Comment by Deantwo** [comment]
> Now we just need to be able to jam ball bearings/screwdrivers/glass shards in grenades

That would be interesting. But would maybe require a new type of grenade, as chem_grenades are used only for chemicals.

**Comment by RemieRichards** [comment]
@Deantwo it's only things listed as "sharp" which at the moment are as follows:
* Circular saws
* Shovels
* Shards
* Broken Bottles
* Fireaxes
* Hatchets
* Throwing stars
* Kitchen Knives
* Scalpels
* The other kind of kitchen knife
* Energy swords

the "size" of the item (it's w_class) simply affects how much damage it does when embedding, getting irritated (just by sitting in the wound it has a chance to do some low but deadly after a while damage) and when falling out

File: html/changelogs/RemieRichards-ExplosionsAndEmbedding.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+  
+author: RemieRichards
+
+delete-after: True
+
+changes: 
+  - rscadd: "Explosions will now blow (throw) items away from the epicenter"
+  - rscadd: "Sharp items can now embedd themselves in a human's limbs"
```
Comment (Deantwo) [inline html/changelogs/RemieRichards-ExplosionsAndEmbedding.yml]: Minor spelling error here?

Comment (RemieRichards) [inline html/changelogs/RemieRichards-ExplosionsAndEmbedding.yml]: yeah that's an error too, if you've seen I corrected all the "embedd"s to embed where they're visible in game and in the code. I don't believe the changelog will matter but if requested to I'll fix the typo.

**Comment by Deantwo** [comment]
Ok cool.

Then my only question is, do windows shatter before or after the shockware that blow (throw) items?
We need more windows on the station now!

**Comment by RemieRichards** [comment]
They shatter after due to the order of Explosion --> Throw items --> Damage Turfs and turf contents
It has to be this order (without a little bit of work) because the Damage Turfs and turf contents section changes a variable that I used.

**Comment by Steelpoint** [comment]
So if you want to remove a embedded item you need to have surgery or hope it eventually falls out? Instead of, you know, pulling it out.

Overall its a good feature, but I think you should be able to yank out a embedded item at the cost of bleeding and some brute damage.

**Comment by RemieRichards** [comment]
The only reason you can't pull it out is "How":
A Verb? That's either a human verb that's permanently there (weird)
An Object verb (Not going to happen in a million years)

**Comment by Deantwo** [comment]
> I think you should be able to yank out a embedded item

Maybe by attacking your self in the limb with a sharp object? A % chance of success though, so you may have to cut your self a lot to get it out.

PS: Why aren't screwdrivers sharp?

**Comment by AlexCrimson** [comment]
The Resist verb seems like the most obvious one to use. We use it for putting out fire and such. What about the Examine thing you get for clicking on yourself? That feature has been pretty useless since the UI was updated to show part-specific damage.

Pulling stuff out should take a few seconds and do more damage to you though. Seeing as pulling shit out like that is very dangerous.

Metal rods so need to be added to the list of things that can stick in you.

**Comment by dannno** [comment]
There definitely needs to be some way to pull stuff out of yourself if this happens.
The 'check yourself for injuries' thing isn't useless unless all of your damage is of one type.

**Comment by AlexCrimson** [comment]
Then Resist seems fine, no?

**Comment by Deantwo** [comment]
Resist doesn't really work for this.
You'll have no control over which item you are trying to pull out unless we make Resist limb based (which would be a first).

I like the thought of stabbing your own arm with a sharp object better. Also gives a reason for the taking damage.

**Comment by GunHog** [comment]
Should you not finish Dismemberment first? I feel this would be an addition to that system rather than standalone.

**Comment by RemieRichards** [comment]
they're completely separate, they don't cross over at all besides "its a thing that happens to limbs"

**Comment by Thunder12345** [comment]
Could make filling with shrapnel a feature of large grenades, since they don't really have any advantage over regular grenades just now

**Comment by paprka** [comment]
I did a little hack to make this throwing knife spin when thrown, it made it look pretty cool

https://github.com/tgstation/-tg-station/pull/7156/files#diff-13349f8d327d30e959f4cac1446c6f44R119

**Comment by Miauw62** [comment]
pls finish dismemberment first

**Comment by dannno** [comment]
>they're completely separate, they don't cross over at all besides "its a thing that happens to limbs"

**Comment by Dorsisdwarf** [comment]
Yeah, increase the price of syndiecards to like 3 TC, give them a really high sticking chance.

**Comment by MrStonedOne** [comment]
Making pulling them out a clickable link in check for injuries seems like the most obvious solution.

**Comment by dannno** [comment]
^ best imo

**Comment by Deantwo** [comment]
I just think link in the chat had the right feel. But if everyone is for it, then sure.
I'll just keep dreaming of people stabbing themselves then.

**Comment by fj45** [comment]
Do not add this for anything but explosions. Paradise had it for a while and we just mostly removed it recently. It made powerful weapon such as the fireaxe damn near useless and turned melee combat into RNG hell.

**Comment by paprka** [comment]
yeah this seems like too much complexity for no real purpose.

I approve of explosions flinging items, but i don't think our medical/damage system needs this much depth

**Comment by Deantwo** [comment]
Well, the added surgery would make more stuff for Medical Doctors to do that isn't chemistry related.

**Comment by paprka** [comment]
this shouldn't only be fixable via surgery

**Comment by dannno** [comment]
spears don't embed why?

**Comment by PKPenguin321** [comment]
>be traitor
>get esword
>hit some assistant in maint
>esword is gone forever as their chest swallows it
>they beat me to death with a screwdriver

no thanks

**Comment by RemieRichards** [comment]
You guys realise items only embed when thrown/blown?
So no, your esword will not disappear unless you're a tool and throw it, in which case you deserve it.

@dannno spears were never "sharp" but I'll add it.

@paprka using the proc to spin something to spin something is a hack? haha. 

@fj45 unless I'm mistaken you seem to be unaware of the first point of it requiring throwing, please correct me otherwise.

@MrStonedOne Yeah that works, surprised I didn't think of a href link.

I'll add embedding to syndicards too.

Apologies with late responses, my sleep cycle is wack so while I should have been awake I was asleep and vice versa :P

**Comment by fj45** [comment]
Yes, embedding when thrown sounds fine.

**Comment by RemieRichards** [comment]
Cool, good to hear.

**Comment by dannno** [comment]
killing someone at low health by throwing your esword into them would be incredible though, i hope that happens to someone

**Comment by dannno** [comment]
speaking of which, there also needs to be a way to remove an item from someone ELSE without surgery
maybe add the embedded items to the strip menu with a decent delay? taking them out can cause more pain to the stripee

**Comment by RemieRichards** [comment]
You can now remove items embedded in yourself at the cost of some damage (8*w_class of the item, so 8,16,24 and if you somehow got a size 4 in their, 32) You also scream because my god that must hurt.
Everyone around you gets a message about how you just ripped it out like a madman.

**Comment by RemieRichards** [comment]
Also just as a heads up, the surgery removes all objects from the target limb, where as ripping the item out only removes that one item. this means overall the surgery is safer and better, but you can rip things out in a pinch.

**Comment by PKPenguin321** [comment]
So just a suggestion but how about make it so you can tear items out of other people as long as they stay still
Maybe put the buttons to tear stuff out in the text when you examine them?
Even without that though this a pretty good PR (i thought it was bad at first since i didn't realize it was on throwing only ¯\_(ツ)_/¯)

**Comment by dannno** [comment]
>add the embedded items to the strip menu with a decent delay

**Comment by Profakos** [comment]
I wonder, could this mechanic open the pathway for future additions, where bullets and syringe gun syringes stick in your limbs? Remove the bullets from the body to forensic find the caliber in a SpaceCSI way.

**Comment by Deantwo** [comment]
SpaceCSI, I like it!
Sadly most weapons use lasers?

Anyway, yeah as @dannno is suggesting, we need a way to remove the embedded items from ~~victims~~friends, also after they are dead.

**Comment by Alek2ander** [comment]
Bleeding should be caused by stuff like this instead of brute damage

**Comment by Deantwo** [comment]
More like, you shouldn't be able to stop the bleeding while this is happening?
Or maybe better, while an item is embedded in you, it should randomly start bleeding every so often.
Meaning that you can stop the bleeding, but it may start again if you don't get the item out.

**Comment by RemieRichards** [comment]
@Alek2ander Bleeding and brute damage are not separate, the amount of blood you bleed out of yourself is tied to the amount of brute damage on your limbs. brute damage and only brute damage.

**Comment by Razharas** [comment]
>We can’t automatically merge this pull request.

**Comment by Razharas** [comment]
Wont throwing items before ex_act() applying make everything movable avoid explosion damage?

**Comment by RemieRichards** [comment]
yes, but it's fine since they should land A: In a person or B: on another turf in the explosion
Also conflict fix coming soon.

**Comment by Razharas** [comment]
Wiat it throws things only in the explosion zone?
That feels strange

**Comment by RemieRichards** [comment]
@Razharas everything within the explosion turfs, so the same ones that would get ex_act()'d, I figure they're the only ones close enough to be moved.

**Comment by Razharas** [comment]
Also i mean like explosion breaks the window and the shard(s) wont fly anywhere because all throwing happened before the explosion effect

**Comment by RemieRichards** [comment]
Oh I... actually don't know why I did that, let me fix it

**Comment by RemieRichards** [comment]
@Razharas Done.

**Comment by Razharas** [comment]
Im ok with this now
Id prefer explosions to throw all movable shit thought but its more for aestetic pleasure than for gameplay

**Comment by RemieRichards** [comment]
well that's kind of out of the scope of this PR (since this is about items embedding, it just so happens to touch explosion code for items)

**Comment by Razharas** [comment]
Is this ready to merge?

**Comment by RemieRichards** [comment]
it was ready 4 days ago.

**Comment by GunHog** [comment]
The urge to make jokes about RR and the wait time for her features is just so hard to resist...

**Comment by dannno** [comment]
merge it all right now
dismemberment included
i will reap the concequences

**Comment by dannno** [comment]
you're welcome

**Comment by GunHog** [comment]
I hope you draw energy from this, @RemieRichards! Let it motivate you to finish dismemberment!!

**Pull request merged**